ARXIVNG-2787: Reduce size of converter image (Create image for each tree) (#168)
arXiv/arxiv-submission-ui: README.md

The idea behind the mock compiler service is to let you work through the entire submission process workflow
without the hassle of installing the compiler service.

## Quick start using arXiv Compiler service (with actual TeX compilation)

There are now multiple converter images that may be configured to compile TeX source.

> Most converter images contains a single TeX tree (to reduce size) along with one

Reviewer: ```suggestion
Most converter images contain a single TeX tree (to reduce size) along with one
```

very large converter image that contains the complete set of all arXiv TeX trees (14.5G).
At the current time you may configure a single converter image to compile TeX sources.

> The Following TeX trees are available: TeX Live 2016 (most recent). TeX Live 2011,

Reviewer: ```suggestion
The following TeX trees are available: TeX Live 2016 (most recent); TeX Live 2011;
```

> TeX Live 2009, teTeX 2, and teTeX 3. The teTeX 2 tree contains multiple local trees.

Reviewer: ```suggestion
TeX Live 2009; teTeX 2; and teTeX 3. The teTeX 2 tree contains multiple local trees.
```

There are two ways to download and install the converter image: manual and automatic.

The 'automated' method lets the compiler service download the converter image for you. If you set the environment
variable 'CONVERTER_IMAGE_PULL' to 1 in the docker-compose.yml file under both the compiler-api: and
the compiler-worker: sections, and then rebuild/up the compiler service, the compiler service will attempt to download
the image from AWS in the event it does not find it locally. There will be a delay during startup while the converter image
is being downloaded.

The second method is to manually download the converter image(s) you need prior to starting up the
submission UI. This will save time during startup.

The 'CONVERTER_IMAGE_PULL' environment variables are now set to 1 (enabled) by default. You may still want to
manually download the image to speed up startup. In the event you accidently remove the converter image the
compiler service will simply download a fresh copy. This seems like a useful safety feature.

In both cases you will need to configure your AWS credentials prior to attempting converter image download.

> To let compiler service download converter image simply set (include version number**)

Reviewer: ```suggestion
In order to let the compiler service download a converter image simply set (including version number**):
```

Author: Yikes!
